builder/module-options.nix

```diff
@@ -0,0 +1,272 @@
+        defaultText = lib.literalMD "Arch derived from {option}`hardware.profile`";
+        example = "aarch64_cortex-a53";
+        description = ''
+          Architecture of board's CPU. 
+
+          :::{.warning}
+          For OpenWRT <19 this is a required field!
+          :::
+        '';
+      };
+
+    };
+
+    packages = {
+      include = lib.mkOption {
+        type = lib.types.listOf lib.types.nonEmptyStr;
+        default = [ ];
+        example = lib.options.literalExpression ''
+          [ 
+            "curl"
+            "iperf3"
+          ]
+        '';
+        description = ''
+          A list of package names to include in the generated image.
+        '';
+      };
+
+      exclude = lib.mkOption {
+        type = lib.types.listOf lib.types.nonEmptyStr;
+        default = [ ];
+        example = lib.options.literalExpression ''
+          [
+            "luci"
+          ]
+        '';
+        description = ''
+          A list of package names to exclude from the generated image.
+        '';
+      };
+    };
+
+    files = lib.mkOption {
+      type = lib.types.listOf fileSubmodule;
+      default = [ ];
+      example = lib.literalExpression ''
+        [ 
+          {
+            source = ./uci-script;
+            target = "/etc/uci-defaults/99-custom";
+          }
+        ]
+      '';
+      description = ''
+        List of files to include in the generated image.
+
+        The path must be either a regular file or a directory of files. 
+
+        :::{.warning}
+        Symlinks are not supported!
+        :::
+      '';
+    };
+
+    disabledServices = lib.mkOption {
+      type = lib.types.listOf lib.types.nonEmptyStr;
+      default = [ ];
+      example = lib.literalExpression ''
+        [
+          "umount"
+        ]
+      '';
+      description = ''
+        A list of services to disable
+      '';
+    };
+
+    image = {
+      extraName = lib.mkOption {
+        type = lib.types.nullOr lib.types.nonEmptyStr;
+        default = null;
+        example = "test-1";
+        description = ''
+          Arbitrary string to add to the generated OpenWRT image file name.
+
+          Specify `null` if you want to use default from OpenWRT imagebuilder
+        '';
+      };
+
+      rootFsPartSize = lib.mkOption {
+        type = lib.types.nullOr lib.types.ints.positive;
+        default = null;
+        example = 8;
+        description = ''
+          Override the default rootfs partition size in MegaBytes
+
+          Specify `null` if you want to use default from OpenWRT imagebuilder
+        '';
+      };
+
+      addLocalKey = lib.mkOption {
+        type = lib.types.nullOr lib.types.bool;
+        default = null;
+        example = false;
+        description = ''
+          Store locally generated signing key in built images
+
+          Specify `null` if you want to use default from OpenWRT imagebuilder
+        '';
+      };
+    };
+
+    sumsFileSha256 = lib.mkOption {
```

Comment:
It's hard to come up with a good name for this one 😅

Reply:
We probably want to use Hash instead of Sha256 for future proofness.